rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -523,16 +523,37 @@ std::shared_ptr<rclcpp::GenericSubscription>
 RecorderImpl::create_subscription(
   const std::string & topic_name, const std::string & topic_type, const rclcpp::QoS & qos)
 {
-  auto subscription = node->create_generic_subscription(
-    topic_name,
-    topic_type,
-    qos,
-    [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message) {
-      if (!paused_.load()) {
-        writer_->write(message, topic_name, topic_type, node->get_clock()->now());
-      }
-    });
-  return subscription;
+  std::function<void(std::shared_ptr<const rclcpp::SerializedMessage>,
+    const rclcpp::MessageInfo &)> recvFun;
+
+  if (record_options_.use_sim_time) {
+    return node->create_generic_subscription(
+      topic_name,
+      topic_type,
+      qos,
+      [this, topic_name, topic_type](std::shared_ptr<const rclcpp::SerializedMessage> message,
+      const rclcpp::MessageInfo & mi) {
+        if (!paused_.load()) {
+          writer_->write(
+            message, topic_name, topic_type, node->now().nanoseconds(),
```

Comment:
Neatpick.
We can use `std::move(message)` when passing the message to the `writer::write(..)` method.